### PR TITLE
add base models to accepted model types. 

### LINKED ISSUE #387: dbt_project_evaluator disagrees with dbt's own best practices guide for base models
I ran into a disagreement between `dbt_project_evaluator` and dbt's own [How we structure our dbt projects](https://docs.getdbt.com/guides/best-practices/how-we-structure/1-guide-overview) guide.

The guide includes two staging `base` models, inside the `models/staging/jaffle_shop/base` directory:

image

These models are named:
- `base_jaffle_shop__customers.sql`
- `base_jaffle_shop__deleted_customers.sql`

If you try this in your own project, `dbt_project_evaluator` will complain that these models should have a `stg_` prefix.

And similarly, `fct_staging_dependent_on_staging` would fail, claiming that `staging` models are depending on other `staging` models.

Given that both of these recommendations come from dbt... can we get some consistency?

### PR DESCRIPTION
**Pull request by dave-connors-3**
This is a:
- [x] bug fix PR with no breaking changes
- [ ] new functionality

## Link to Issue 
Closes #387 

## Description & motivation

Adds base models as an acceptable model type. 

Chose to exclude the new model i added to the integration tests so as not to have to edit all the test and documentation related seeds, but validated the output in `int_all_graph_resources`:

image

I also added an accepted values test to make sure the types match the variable -- not a perfect test, but at least protects against nulls and/or some faulty logic

## Checklist
- [x] I have verified that these changes work locally on the following warehouses (Note: it's okay if you do not have access to all warehouses, this helps us understand what has been covered)
    - [ ] BigQuery
    - [ ] Postgres
    - [ ] Redshift
    - [ ] Snowflake
    - [ ] Databricks
    - [ ] DuckDB
    - [ ] Trino/Starburst
- [ ] I have updated the README.md (if applicable)
- [ ] I have added tests & descriptions to my models (and macros if applicable)

### CODE REVIEW

File: integration_tests/models/dbt_project_evaluator_schema_tests/core.yml [hunk 1]
```diff
@@ -21,6 +21,10 @@ models:
         description: this contains the on_schema_change setting for incremental models. This column was sometimes an empty string, so should be tested to detect regressions
         tests:
           - is_not_empty_string
+      - name: model_type
+        tests: 
+          - accepted_values:
+              values: "{{ var('model_types') }}"
```
Comment (graciegoheen) [inline integration_tests/models/dbt_project_evaluator_schema_tests/core.yml:27]: Won't this also include 'extra' and 'new_model_type'? Is that desired?

File: integration_tests/dbt_project.yml [hunk 2]
```diff
@@ -61,7 +61,7 @@ vars:
   # ensure integration tests run successfully when there are 0 of a given model type (extra)
   exclude_packages: ['exclude_package']
   exclude_paths_from_project: ["/to_exclude/","source_3.table_6"]
-  model_types: ['staging', 'intermediate', 'marts', 'other', 'extra', 'new_model_type']
+  model_types: ['base','staging', 'intermediate', 'marts', 'other', 'extra', 'new_model_type']
```
Comment (graciegoheen) [inline integration_tests/dbt_project.yml:64]: This is just for the integration test right?

Comment (dave-connors-3) [reply integration_tests/dbt_project.yml:64]: yep -- that's what happens when you commit from inside a subdirectory 🤦

**Comment by graciegoheen** [comment]
> Chose to exclude the new model i added to the integration tests so as not to have to edit all the test and documentation related seeds, but validated the output in int_all_graph_resources

sounds like a good use case for unit testing.... hahaha

**Review by graciegoheen — APPROVED** [review]
LGTM

**Pull request merged**